tests/agent_server/test_resource_sets.py

```diff
@@ -93,3 +94,915 @@ async def test_resource_sets_via_put_version(server, client, environment, client
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key8",
+            "value": "8",
+            "id": "test::Resource[agent1,key=key8],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+    ]
+    resource_sets = {
+        "test::Resource[agent1,key=key1]": "set-a",
+        "test::Resource[agent1,key=key2]": "set-a",
+        "test::Resource[agent1,key=key3]": "set-b",
+        "test::Resource[agent1,key=key4]": "set-b",
+        "test::Resource[agent1,key=key7]": "set-c",
+        "test::Resource[agent1,key=key8]": "set-c",
+    }
+
+    result = await client.put_version(
+        tid=environment,
+        version=version,
+        resources=resources,
+        resource_state={},
+        unknowns=[],
+        version_info={},
+        compiler_version=get_compiler_version(),
+        resource_sets=resource_sets,
+    )
+    assert result.code == 200
+    version = await clienthelper.get_version()
+    resources_partial = [
+        {
+            "key": "key1",
+            "value": "100",
+            "id": "test::Resource[agent1,key=key1],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key2",
+            "value": "200",
+            "id": "test::Resource[agent1,key=key2],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key9",
+            "value": "900",
+            "id": "test::Resource[agent1,key=key9],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key91",
+            "value": "910",
+            "id": "test::Resource[agent1,key=key91],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key92",
+            "value": "920",
+            "id": "test::Resource[agent1,key=key92],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+    ]
+
+    result = await client.put_partial(
+        tid=environment,
+        version=version,
+        resources=resources_partial,
+        resource_state={},
+        unknowns=[],
+        version_info=None,
+        compiler_version=get_compiler_version(),
+        resource_sets={
+            "test::Resource[agent1,key=key1]": "set-a",
+            "test::Resource[agent1,key=key2]": "set-a",
+            "test::Resource[agent1,key=key3]": "set-b",
+            "test::Resource[agent1,key=key4]": "set-b",
+            "test::Resource[agent1,key=key91]": "set-f",
+            "test::Resource[agent1,key=key92]": "set-f",
+        },
```

Comment:
The `resource_sets` argument should only contains entries for the resource part of the partial compile.
```suggestion
        resource_sets={
            "test::Resource[agent1,key=key1]": "set-a",
            "test::Resource[agent1,key=key2]": "set-a",
            "test::Resource[agent1,key=key91]": "set-f",
            "test::Resource[agent1,key=key92]": "set-f",
        },
```